app/src/main/java/com/eveningoutpost/dexdrip/AndroidFileNamePicker.java

```diff
@@ -0,0 +1,385 @@
+package com.eveningoutpost.dexdrip;
+// Based on https://stackoverflow.com/questions/13209494/how-to-get-the-full-file-path-from-uri/60642994#answer-60642994
```

Comment:
Did you check the license or is this file completely written by you?

Reply:
The file is copied almost as is from stackoverflow.
As far as I can tell, we are doing a fair use of this code. I'm based on answers like [this](https://www.ictrecht.nl/en/blog/what-is-the-license-status-of-stackoverflow-code-snippets).
Obviously, there are other opinions on the internet. I'm not a lower, but I fill that in a GPL project there shouldn't be a problem to use it.